CHANGELOG.md

```diff
@@ -1,3 +1,5 @@
+<!-- markdownlint-disable MD024 -->
```

Comment:
Your logic for the various adjustments to lints seems reasonable.

I suggest stating that reasoning in a comment above (or below) each of these adjustments.  The advantage of including the reasoning here locally is that folks are much more likely to read and understand the reasoning, rather than beginning cargo-culting (hmm, that other file disables the check, so let's disable it everywhere...:facepalm:)
 
I've included an explicit "change" suggestion here as an example of what I mean, but it applies to all the rest of the places as well (I won't comment on the rest of them).

```suggestion
<!-- MD024 - We want repeated headings in a changelog file -->
<!-- markdownlint-disable MD024 -->
```